crates/oxc_codegen/src/gen.rs

```diff
@@ -92,8 +92,13 @@ impl<const MINIFY: bool> Gen<MINIFY> for Hashbang {
 
 impl<const MINIFY: bool> Gen<MINIFY> for Directive {
     fn gen(&self, p: &mut Codegen<{ MINIFY }>, _ctx: Context) {
-        // Use the string value instead of the raw self.directive because it can cannot escaped values.
-        print_str(self.expression.value.as_str(), p);
+        // A Use Strict Directive may not contain an EscapeSequence or LineContinuation.
+        // So here should print original `directive` value, the `expression` value is escaped str.
+        // See https://github.com/babel/babel/blob/main/packages/babel-generator/src/generators/base.ts#L64
+        let quote = choose_quote(self.directive.as_str());
+        p.print(quote as u8);
+        p.print_str(self.directive.as_str().to_string().as_bytes());
```

Comment:
`to_string` allocates, can we remove it somehow?

Reply:
Updated.